tests/tls_lib_test.sh

```diff
@@ -171,7 +171,15 @@ else
 	echo "*** WolfSSL sample test FAILED"
 fi
 
-if [ -n "${openssl_success}" ] && [ -n "${mbedtls_success}" ] && [ -n "${wolfssl_success}" ]; then
+echo "*** Testing Paho MQTT sample"
+if [ ! -d ${CLOUD_SUPPORT_DIR}/paho-openssl/libs/paho.mqtt.embedded-c ]; then
+  mkdir -p ${CLOUD_SUPPORT_DIR}/paho-openssl/libs/
+  git clone https://github.com/eclipse/paho.mqtt.embedded-c/ ${CLOUD_SUPPORT_DIR}/paho-openssl/libs/paho.mqtt.embedded-c
+fi
+build_sample ${CLOUD_SUPPORT_DIR}/paho-openssl ${TOB_LIB_DIR} ${TOB_INC_DIR} && paho_openssl_success=1
```

Comment:
This is a compilation-only test.  To truly test the functionality, we will need to spin up a TLS enabled MQTT broker  configured for certificate authentication.

The main thing Trust Onboard related that this relies on, aside from successful compilation, is the OpenSSL engine which is tested above.